Add Account Notes Feature

douglasnobree opened the pull request:
<html>
<body>
<!--StartFragment--><h2 data-start="90" data-end="115">✨ Account Notes System</h2>
<p data-start="117" data-end="260">This PR introduces a complete <strong data-start="147" data-end="164">account notes</strong> feature to the Roblox Account Manager, allowing users to better organize and identify accounts.</p>
<hr data-start="262" data-end="265">
<h3 data-start="267" data-end="289"><strong data-start="271" data-end="289">Features Added</strong></h3>
<h4 data-start="291" data-end="316">1. <strong data-start="299" data-end="316">Account Notes</strong></h4>
<ul data-start="317" data-end="541">
<li data-start="317" data-end="366">
<p data-start="319" data-end="366">Users can now attach custom notes to accounts</p>
</li>
<li data-start="367" data-end="437">
<p data-start="369" data-end="437">Notes appear next to the username in the list:<br data-start="415" data-end="418">
<code data-start="420" data-end="437">username • note</code></p>
</li>
<li data-start="438" data-end="487">
<p data-start="440" data-end="487">Notes are <strong data-start="450" data-end="463">encrypted</strong> along with account data</p>
</li>
<li data-start="488" data-end="541">
<p data-start="490" data-end="541">Easy-to-use <strong data-start="502" data-end="515">Edit Note</strong> dialog (multi-line input)</p>
</li>
</ul>
<h4 data-start="543" data-end="585">2. <strong data-start="551" data-end="585">Backend (<code data-start="562" data-end="582">account_manager.py</code>)</strong></h4>
<ul data-start="586" data-end="834">
<li data-start="586" data-end="632">
<p data-start="588" data-end="632">Added <code data-start="594" data-end="600">note</code> field to account data structure</p>
</li>
<li data-start="633" data-end="713">
<p data-start="635" data-end="641">Added:</p>
<ul data-start="644" data-end="713">
<li data-start="644" data-end="680">
<p data-start="646" data-end="680"><code data-start="646" data-end="680">set_account_note(username, note)</code></p>
</li>
<li data-start="683" data-end="713">
<p data-start="685" data-end="713"><code data-start="685" data-end="713">get_account_note(username)</code></p>
</li>
</ul>
</li>
<li data-start="714" data-end="779">
<p data-start="716" data-end="779">Added <strong data-start="722" data-end="745">automatic migration</strong> to add notes to existing accounts</p>
</li>
<li data-start="780" data-end="834">
<p data-start="782" data-end="834">Fully integrated with the existing encryption system</p>
</li>
</ul>
<h4 data-start="836" data-end="860">3. <strong data-start="844" data-end="860">UI (<code data-start="850" data-end="857">ui.py</code>)</strong></h4>
<ul data-start="861" data-end="1193">
<li data-start="861" data-end="912">
<p data-start="863" data-end="912"><code data-start="863" data-end="883">refresh_accounts()</code> now displays username + note</p>
</li>
<li data-start="913" data-end="974">
<p data-start="915" data-end="974"><code data-start="915" data-end="940">get_selected_username()</code> updated to handle formatted names</p>
</li>
<li data-start="975" data-end="1020">
<p data-start="977" data-end="1020">Added <strong data-start="983" data-end="996">Edit Note</strong> button to Quick Actions</p>
</li>
<li data-start="1021" data-end="1193">
<p data-start="1023" data-end="1056">Created note editing dialog with:</p>
<ul data-start="1059" data-end="1193">
<li data-start="1059" data-end="1081">
<p data-start="1061" data-end="1081">Current note preview</p>
</li>
<li data-start="1084" data-end="1106">
<p data-start="1086" data-end="1106">Multi-line text area</p>
</li>
<li data-start="1109" data-end="1132">
<p data-start="1111" data-end="1132">Save / Cancel buttons</p>
</li>
<li data-start="1135" data-end="1161">
<p data-start="1137" data-end="1161">Consistent dark theme UI</p>
</li>
<li data-start="1164" data-end="1193">
<p data-start="1166" data-end="1193">Centered window positioning</p>
</li>
</ul>
</li>
</ul>
<h4 data-start="1195" data-end="1226">4. <strong data-start="1203" data-end="1226">Layout Improvements</strong></h4>
<ul data-start="1227" data-end="1536">
<li data-start="1227" data-end="1274">
<p data-start="1229" data-end="1274">Main window height increased: <code data-start="1259" data-end="1274">500px → 580px</code></p>
</li>
<li data-start="1275" data-end="1316">
<p data-start="1277" data-end="1316">Reduced button font size for better fit</p>
</li>
<li data-start="1317" data-end="1358">
<p data-start="1319" data-end="1358">Recent games list reduced: <code data-start="1346" data-end="1358">6 → 5 rows</code></p>
</li>
<li data-start="1359" data-end="1476">
<p data-start="1361" data-end="1408">Reorganized bottom buttons into a <strong data-start="1395" data-end="1407">2x2 grid</strong>:</p>
<pre class="overflow-visible!" data-start="1411" data-end="1476"><div class="contain-inline-size rounded-2xl relative bg-token-sidebar-surface-primary"><div class="sticky top-9"><div class="absolute end-0 bottom-0 flex h-9 items-center pe-2"><div class="bg-token-bg-elevated-secondary text-token-text-secondary flex items-center gap-4 rounded-sm px-2 font-sans text-xs"></div></div></div><div class="overflow-y-auto p-4" dir="ltr"><code class="whitespace-pre!"><span><span><span class="hljs-variable">Add</span></span><span> </span><span><span class="hljs-variable">Account</span></span><span>     </span><span><span class="hljs-operator">|</span></span><span> </span><span><span class="hljs-built_in">Remove</span></span><span>
</span><span><span class="hljs-built_in">Launch</span></span><span> </span><span><span class="hljs-variable">Browser</span></span><span>  </span><span><span class="hljs-operator">|</span></span><span> </span><span><span class="hljs-variable">Settings</span></span><span>
</span></span></code></div></div></pre>
</li>
<li data-start="1477" data-end="1536">
<p data-start="1479" data-end="1536">All buttons now expand horizontally properly (<code data-start="1525" data-end="1535">fill="x"</code>)</p>
</li>
</ul>
<hr data-start="1538" data-end="1541">
<h3 data-start="1543" data-end="1565"><strong data-start="1547" data-end="1565">Modified Files</strong></h3>
<div class="_tableContainer_1rjym_1"><div tabindex="-1" class="group _tableWrapper_1rjym_13 flex w-fit flex-col-reverse">
File | Description
-- | --
account_manager.py | Added note support, migration system, encryption integration
ui.py | Display notes, new dialog, layout fixes, input handling

</div></div>
<hr data-start="1773" data-end="1776">
<h3 data-start="1778" data-end="1800"><strong data-start="1782" data-end="1800">UI/UX Benefits</strong></h3>
<ul data-start="1801" data-end="2003">
<li data-start="1801" data-end="1855">
<p data-start="1803" data-end="1855">Notes help distinguish multiple accounts more easily</p>
</li>
<li data-start="1856" data-end="1907">
<p data-start="1858" data-end="1907">Bullet separator provides clean visual separation</p>
</li>
<li data-start="1908" data-end="1958">
<p data-start="1910" data-end="1958">Interfaces remain consistent with the dark theme</p>
</li>
<li data-start="1959" data-end="2003">
<p data-start="1961" data-end="2003">Layout feels less cramped and more aligned</p>
</li>
</ul>
<hr data-start="2005" data-end="2008">
<h3 data-start="2010" data-end="2040"><strong data-start="2014" data-end="2040">Backward Compatibility</strong></h3>
<ul data-start="2041" data-end="2143">
<li data-start="2041" data-end="2091">
<p data-start="2043" data-end="2091">✅ Existing accounts are automatically migrated</p>
</li>
<li data-start="2092" data-end="2110">
<p data-start="2094" data-end="2110">✅ No data loss</p>
</li>
<li data-start="2111" data-end="2143">
<p data-start="2113" data-end="2143">✅ Empty notes handled safely</p>
</li>
</ul>
<hr data-start="2145" data-end="2148">
<h3 data-start="2150" data-end="2175"><strong data-start="2154" data-end="2175">Testing Checklist</strong></h3>
<ul class="contains-task-list" data-start="2176" data-end="2469">
<li class="task-list-item" data-start="2176" data-end="2209">
<p data-start="2182" data-end="2209"><input disabled="" type="checkbox" checked=""> Add notes to new accounts</p>
</li>
<li class="task-list-item" data-start="2210" data-end="2249">
<p data-start="2216" data-end="2249"><input disabled="" type="checkbox" checked=""> Edit notes of existing accounts</p>
</li>
<li class="task-list-item" data-start="2250" data-end="2287">
<p data-start="2256" data-end="2287"><input disabled="" type="checkbox" checked=""> Notes persist across sessions</p>
</li>
<li class="task-list-item" data-start="2288" data-end="2346">
<p data-start="2294" data-end="2346"><input disabled="" type="checkbox" checked=""> Notes encrypt correctly when encryption is enabled</p>
</li>
<li class="task-list-item" data-start="2347" data-end="2384">
<p data-start="2353" data-end="2384"><input disabled="" type="checkbox" checked=""> Migration runs without errors</p>
</li>
<li class="task-list-item" data-start="2385" data-end="2418">
<p data-start="2391" data-end="2418"><input disabled="" type="checkbox" checked=""> UI buttons align properly</p>
</li>
<li class="task-list-item" data-start="2419" data-end="2469">
<p data-start="2425" data-end="2469"><input disabled="" type="checkbox" checked=""> Account selection works with notes present</p>
</li>
</ul>
<hr data-start="2471" data-end="2474">
<h3 data-start="2476" data-end="2494"><strong data-start="2480" data-end="2494">How to Use</strong></h3>
<ol data-start="2495" data-end="2589">
<li data-start="2495" data-end="2529">
<p data-start="2498" data-end="2529">Select an account from the list</p>
</li>
<li data-start="2530" data-end="2552">
<p data-start="2533" data-end="2552">Click <strong data-start="2539" data-end="2552">Edit Note</strong></p>
</li>
<li data-start="2553" data-end="2571">
<p data-start="2556" data-end="2571">Enter note text</p>
</li>
<li data-start="2572" data-end="2589">
<p data-start="2575" data-end="2589">Click <strong data-start="2581" data-end="2589">Save</strong></p>
</li>
</ol>
<!--EndFragment-->
</body>
</html>

douglasnobree committed:
Add Account Notes Feature

evanovar commented:
I really like the idea of adding the note feature, it’s a great addition, and I’d be happy to merge it once a few small UI tweaks are made

Some feedback:
1. **Bottom button layout:** The new 2x2 grid feels too large, I’d prefer keeping the layout more compact.
2. **Main window height:** Increased from 500px → 580px , this feels too tall; around 520px works best.
3. **Note UI width:** It looks a bit too narrow on my screen, please increase it slightly for better readability.

Once those adjustments are made, I'll merge it!

douglasnobree committed:
fix: adjust UI dimensions and layout for better UX

- Reduce main window height from 580px to 520px for more compact layout
- Restore bottom buttons to single horizontal row (4 buttons side-by-side)
- Increase Edit Note dialog width from 400px to 450px for better readability
- Increase Edit Note dialog height from 200px to 220px to prevent button text cutoff
- Reduce note text area height from 4 to 3 lines to accommodate larger button spacing

douglasnobree commented:
@evanovar done sir
